hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/processor/SourceProcessors.java

```diff
@@ -421,4 +434,80 @@ public static ProcessorMetaSupplier streamFilesP(
             }
         };
     }
+
+    /**
+     * Returns a supplier of processors for a source that the user can create
+     * using the {@link com.hazelcast.jet.pipeline.SourceBuilder}. This variant
+     * creates a source that emits items without timestamps.
+     *
+     * @param createFn function that creates the source's state object
+     * @param fillBufferFn function that fills Jet's buffer with items to emit
+     * @param destroyFn function that cleans up the resources held by the state object
+     * @param preferredLocalParallelism preferred local parallelism of the source vertex
+     * @param <S> type of the source's state object
+     * @param <T> type of items the source emits
+     */
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    public static <S, T> ProcessorMetaSupplier convenientSourceP(
+            @Nonnull DistributedFunction<? super Processor.Context, ? extends S> createFn,
+            @Nonnull DistributedBiConsumer<? super S, ? super SourceBuffer<T>> fillBufferFn,
+            @Nullable DistributedConsumer<? super S> destroyFn,
+            int preferredLocalParallelism
+    ) {
+        checkSerializable(createFn, "createFn");
+        checkSerializable(fillBufferFn, "fillBufferFn");
+        checkSerializable(destroyFn, "destroyFn");
+        checkNotNegative(preferredLocalParallelism, "preferredLocalParallelism must not be negative");
```

Comment:
preferredLocalParallelism can be -1 or >=1. It should also be better documented

Reply:
You can also document and enforce it in `ProcessorMetaSupplier.preferredLocalParallelism`. Check function is in `Vertex.checkLocalParallelism`.

Reply:
Just noticed it can also be 0 in this method, meaning that total parallelism of 1 should be used. We should document that.